README.md

```diff
@@ -1,44 +1,284 @@
 # Seccompiler
 
-Provides easy-to-use linux [seccomp-bpf](https://www.kernel.org/doc/Documentation/prctl/seccomp_filter.txt)
-jailing for rust-based VMMs.
+Provides easy-to-use linux [seccomp-bpf](https://www.kernel.org/doc/html/latest/userspace-api/seccomp_filter.html) jailing.
+
+Seccomp is a safety net that can be applied on a process, to restrict a
+malicious user, that found a way of running arbitrary code, from executing
+potentially dangerous system calls.
+
+Writing BPF programs by hand is difficult and error-prone. This crate provides
+a library of high-level wrappers over system call filtering, as well as
+functions for generating and installing the filters.
+
+## Supported platforms
+
+Due to the fact that seccomp is a linux-specific feature, this crate is
+supported only on linux systems.
+
+Supported host arhitectures:
+- Little-endian x86_64
```

Comment:
A couple of things.

In regards to architecture, the system call numbers are different. When expressing JSON filters, the user specifies the syscall name, that is translated by seccompiler under the hood, to the arch-specific number.
Even if the user is not using the JSON functionality, seccomp-bpf filters are recommended to embed an architecture validation step at the beginning, to prevent the filter from being installed on a different arch (which will inherently have different syscall numbers).

And in regards to Big Endian, the BPF filter would be different. For example, if you want to compare an argument against a 64-bit integer, you need to split the integer into most significant and least significant halves, because the BPF has a 32-bit auxiliary register.

Reply:
There are definitely ways of making the crate available on multiple architectures, but I think we should do this on a per-need basis, since it has a dev and maintenance cost

Reply:
you mean having a `#![cfg(target_endian = "little")]` in the code? If so, will definitely do that in a future PR.

Reply:
Yap.